py/tests/plane_move.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/python3
+
+import pykms
+import random
+import time
+import sys
+import select
+import selectors
+
+black = pykms.RGB(0, 0, 0)
+
+card = pykms.Card()
+
+if not card.has_atomic:
+    print('Atomic mode settings is not supported :(')
+    sys.exit()
+
+res = pykms.ResourceManager(card)
+
+conn_list = []
+crtc_list = []
+mode_list = []
+rootplane_list = []
+fb_list = []
+colors = []
+
+src_w = 300
+src_h = 300
+
+# get the list of connectors
+for conn in card.connectors:
+    if conn.connected() == 1:
+        conn_list.append(conn)
+
+if len(conn_list) != 2:
+    print('We need two connectors')
+    sys.exit()
+
+card.disable_planes()
+
+print('Have {} connected connectors:'.format(len(conn_list)))
+for conn in conn_list:
+    crtc = res.reserve_crtc(conn)
+    crtc_list.append(crtc)
+
+    mode = conn.get_default_mode()
+    mode_list.append(mode)
+
+    fb_tmp = pykms.DumbFramebuffer(card, src_w, src_h, 'XR24');
+    fb_list.append(fb_tmp)
+
+    rootplane = res.reserve_primary_plane(crtc, pykms.PixelFormat.XRGB8888)
+    rootplane_list.append(rootplane)
+
+    print(' {}: {} ({}x{})'.format(conn.idx, conn.fullname,
+        mode.hdisplay, mode.vdisplay))
+
+for i in range(2):
+    r = random.randrange(255)
+    g = random.randrange(255)
+    b = random.randrange(255)
+    color = pykms.RGB(r, g, b)
+    pykms.draw_rect(fb_list[i], 0, 0, src_w, src_h, color)
+
+for i in range(2):
+    req = pykms.AtomicReq(card)
+    modeb = mode_list[i].to_blob(card)
+    req.add(conn_list[i], 'CRTC_ID', crtc_list[i].id)
+    req.add(crtc_list[i], {'ACTIVE': 1,
+                    'MODE_ID': modeb.id})
+    req.add(rootplane_list[i], {'FB_ID': fb_list[i].id,
+                'CRTC_ID': crtc_list[i].id,
+                'SRC_W': src_w << 16,
+                'SRC_H': src_h << 16,
+                'CRTC_W': src_w,
+                'CRTC_H': src_h})
+
+    req.commit_sync(allow_modeset = True)
+
+input("You have one box on each display, ENTER to continue\n")
+
+req = pykms.AtomicReq(card)
+req.add(rootplane_list[0], {"FB_ID": 0,
+                "CRTC_ID": 0})
+req.commit_sync(allow_modeset = True)
+
+req = pykms.AtomicReq(card)
+req.add(rootplane_list[0], {'FB_ID': fb_list[0].id,
+            'CRTC_ID': crtc_list[1].id,
+            'SRC_W': src_w << 16,
+            'SRC_H': src_h << 16,
+            'CRTC_X': 150,
+            'CRTC_Y': 150,
+            'CRTC_W': src_w,
+            'CRTC_H': src_h})
+req.commit_sync(allow_modeset = True)
+
+input("The box from crtc0 is moved underneath the box on crtc1, ENTER to continue\n")
+
```

Comment:
If you use predefined colors, you can use "red box" etc in the prints. Also, you can use the connector names instead of crtc0. That way it's much easier to read the text and see that yes, the displays are correct.